pkg/clusters/addons/kong/builder.go

```diff
@@ -36,6 +41,34 @@ func (b *Builder) WithPostgreSQL() *Builder {
 	return b
 }
 
+// WithEnterprise deploying kong enterpise
+func (b *Builder) WithEnterprise() *Builder {
+	b.enterprise = true
+	b.repo = DefaultEnterpriseImageRepo
+	b.tag = DefaultEnterpriseImageTag
+	b.license = KongEnterpriseTestLicense
+	b.kongAdminPassword = EnterpriseKongAdminDefaultPWD
+	return b
+}
+
+// WithImage specify docker image repo and tag
+func (b *Builder) WithImage(repo, tag string) *Builder {
+	b.repo = repo
+	b.tag = tag
+	return b
+}
+
+// WithLicense specify license secret name
+func (b *Builder) WithLicense(license string) *Builder {
+	b.license = license
+	return b
+}
```

Comment:
Similar to previous comments the naming here is ambiguous, we should add specificity:

```suggestion
// WithEnterpriseLicenseSecretName configures the name of the Kubernetes Secret object that will
// be used to store and manage the Kong Enterprise license data.
func (b *Builder) WithEnterpriseLicenseSecretName(secretName string) *Builder {
	b.license = secretName
	return b
}
```

Reply:
This one is no longer valid, since we've opted to not take secret name as an option :+1:

Reply:
41d7c75fb3875bdbcd50ff3757f5e4ac8c62bb78